### PR TITLE
Added retrieving size of search marks in mercator

### PR DESCRIPTION
**Pull request by rokuz**
https://jira.mail.ru/browse/MAPSME-2193

### CODE REVIEW

File: drape_frontend/message_subclasses.hpp [hunk 1]
```diff
@@ -818,4 +818,28 @@ class SetDisplacementModeMessage : public Message
   int m_mode;
 };
 
+class RequestSymbolsSizeMessage : public Message
+{
+public:
+  explicit RequestSymbolsSizeMessage(vector<string> const & symbols,
+                                     function<void(vector<m2::PointU> const &)> const & callback)
```
Comment (darina) [inline drape_frontend/message_subclasses.hpp]: Может, обернуть в тип?

File: map/framework.cpp [hunk 2]
```diff
@@ -2588,6 +2600,16 @@ void Framework::BlockTapEvents(bool block)
   CallDrapeFunction(bind(&df::DrapeEngine::BlockTapEvents, _1, block));
 }
 
+m2::PointD Framework::GetSearchMarkSize(SearchMarkType searchMarkType)
+{
+  if (m_searchMarksSizes.empty())
```
Comment (darina) [inline map/framework.cpp]: можно проверить так:
static_cast<size_t>(searchMarkType) < m_searchMarksSizes.size()

Comment (rokuz) [inline map/framework.cpp]: Добавлен ASSERT

File: drape_frontend/backend_renderer.cpp [hunk 3]
```diff
@@ -274,6 +274,23 @@ void BackendRenderer::AcceptMessage(ref_ptr<Message> message)
       m_readManager->Allow3dBuildings(msg->Allow3dBuildings());
       break;
     }
+  case Message::RequestSymbolsSize:
+    {
+      ref_ptr<RequestSymbolsSizeMessage> msg = message;
+      auto const & symbols = msg->GetSymbols();
+
+      vector<m2::PointU> sizes;
```
Comment (ygorshenin) [inline drape_frontend/backend_renderer.cpp]: Why not to:
```c++
vector<m2::PointU> sizes(symbols.size());
for (size_t i = 0; i < symbols.size(); ++i)
{
  ...
  sizes[i] = region.GetPixelSize();
}
```
?

File: drape_frontend/drape_engine.hpp [hunk 4]
```diff
@@ -148,6 +148,10 @@ class DrapeEngine
 
   void SetDisplacementMode(int mode);
 
+  using TRequestSymbolsSizeCallback = function<void(vector<m2::PointU> const &)>;
+  void RequestSymbolsSize(vector<string> const & symbols,
```
Comment (ygorshenin) [inline drape_frontend/drape_engine.hpp]: Please, insert a blank line between using and method declarations.

File: map/user_mark.hpp [hunk 5]
```diff
@@ -50,6 +50,14 @@ class UserMark : public df::UserPointMark
   mutable UserMarkContainer * m_container;
 };
 
+enum SearchMarkType
+{
+  DefaultSearchMark = 0,
```
Comment (ygorshenin) [inline map/user_mark.hpp]: By style guide the naming should be:
```
SEARCH_MARK_TYPE_DEFAULT,
SEARCH_MARK_TYPE_BOOKING,
SEARCH_MARK_TYPE_COUNT
```
but I'm not insisting.

File: drape_frontend/message_subclasses.hpp [hunk 6]
```diff
@@ -818,4 +818,30 @@ class SetDisplacementModeMessage : public Message
   int m_mode;
 };
 
+class RequestSymbolsSizeMessage : public Message
+{
+public:
+  using TRequestSymbolsSizeCallback = function<void(vector<m2::PointU> const &)>;
+
+  explicit RequestSymbolsSizeMessage(vector<string> const & symbols,
```
Comment (ygorshenin) [inline drape_frontend/message_subclasses.hpp]: No need in ```explicit``` for multiple-argument ctors.

**Comment by darina** [comment]
LGTM

**Comment by rokuz** [comment]
PTAL @ygorshenin

**Comment by ygorshenin** [comment]
LGTM

**Comment by ygorshenin** [comment]
Thanks for doing that!

**Pull request merged**